sound/soc/sof/sof-priv.h

```diff
@@ -395,6 +395,12 @@ struct snd_sof_dev {
 
 	/* power states related */
 	enum sof_d0_substate d0_substate;
+	/*
+	 * Set this flag which will be used to ignore the dapm PM events
+	 * and the suspending of d0i3_compatible stream during the S0ix
+	 * suspend/resume cycle, and clear the flag once resuming finished.
+	 */
+	bool s0_suspend;
```

Comment:
@keyonjie what do you meanby "shared to different drivers" in the commit message. Let just simply say that the flag will be used to indicate that the system is entering S0ix. Even in the comment above, how and where it is used is not needed here.

Reply:
Agree with @ranj063 the comment should be something like
'flag set in the "prepare" suspend stage to track the intended power target (S0 idle or S3)